lib/src/boollist.dart

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:collection' show ListMixin;
+import 'dart:typed_data' show Uint32List;
+
+import 'unmodifiable_wrappers.dart' show NonGrowableListMixin;
+
+/// A BoolList implementation for Dart to store boolean values.
+///
+/// Uses list of integers as internal storage to reduce memory usage.
+///
+/// Internal storage of the growable [BoolList] is dynamically changed in the
+/// following cases:
```

Comment:
I wouldn't document the exact algorithm for growing and shrinking. It's better to be able to tweak the limits without anyone complaining.